framework/wazuh/syscheck.py

```diff
@@ -82,24 +82,31 @@ def clear(agent_list=None):
     result = AffectedItemsWazuhResult(all_msg='Syscheck database was cleared on returned agents',
                                       some_msg='Syscheck database was not cleared on some agents',
                                       none_msg="No syscheck database was cleared")
-    wdb_conn = WazuhDBConnection()
 
     system_agents = get_agents_info()
-    for agent in agent_list:
-        if agent not in system_agents:
-            result.add_failed_item(id_=agent, error=WazuhResourceNotFound(1701))
+    wdb_conn = None
+    for agent_id in agent_list:
+        if agent_id not in system_agents:
+            result.add_failed_item(id_=agent_id, error=WazuhResourceNotFound(1701))
         else:
-            try:
-                wdb_conn.execute("agent {} sql delete from fim_entry".format(agent), delete=True)
-                # Update key fields which contains keys to value 000
-                wdb_conn.execute("agent {} sql update metadata set value = '000' "
-                                 "where key like 'fim_db%'".format(agent), update=True)
-                wdb_conn.execute("agent {} sql update metadata set value = '000' "
-                                 "where key = 'syscheck-db-completed'".format(agent), update=True)
-                result.affected_items.append(agent)
-            except WazuhError as e:
-                result.add_failed_item(id_=agent, error=e)
-
+            agent = Agent(agent_id)
+            agent.load_info_from_db()
```

Comment:
This will open a new connection to the DB for each agent that is iterated, which can be quite slow. Although the endpoint belongs to Experimental and it is assumed that these types of problems can occur, I would recommend reviewing other similar cases in the Wazuh framework to see how this problem is dealt with. 

An example (I don't know if it could be applied in this case) is to obtain the version and ID of agent_list in global.db in one go, and then iterate them without needing to open a connection for each one. You can take a look at a similar implementation that I did here: 
https://github.com/wazuh/wazuh/blob/c74fdf8d350bd7c8bbfee30a26d87dc945a44f5d/framework/wazuh/agent.py#L971-L976

Evaluate if it would be worth it.

Reply:
I changed it as you suggested here dcdb11746a0d8bdb0ffe029be0d5774c590f2969